ffi/src/primitives.rs

```diff
@@ -21,8 +22,8 @@ impl<'itm> TryFromReprC<'itm> for bool {
     type Source = <u8 as TryFromReprC<'itm>>::Source;
     type Store = ();
 
-    unsafe fn try_from_repr_c(source: Self::Source, _: &mut ()) -> Result<Self> {
-        let source: u8 = TryFromReprC::try_from_repr_c(source, &mut ())?;
+    unsafe fn try_from_repr_c(source: Self::Source, store: &mut ()) -> Result<Self> {
```

Comment:
In this case should we specify `type Store = <u8 as TryFromReprC<'itm>>::Store`?

In other places as well.

Reply:
no, why would we? the ownership is returned from the function

Reply:
Because we recovering `bool` from `u8` so i think it is reasonable to explicitly state this for `Store` as well since we reuse store from function argument.

Reply:
isn't `<u8 as TryFromReprC<'itm>>::Store` == `()`. Otherwise, this wouldn't compile